cns/restserver/util.go

```diff
@@ -40,6 +46,49 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SetPnpIDMacaddressMapping() error {
+	service.Lock()
+	defer service.Unlock()
+	VfMacAddressMapping, err := fetchMacAddressPnpIDMapping()
+	if err != nil {
+		return err
+	}
+	service.PnpIDByMacAddress = VfMacAddressMapping
+	return nil
+}
+
+func (service *HTTPRestService) getPNPIDFromMacAddress(macAddress string) (string, error) {
+	if _, ok := service.PnpIDByMacAddress[macAddress]; !ok {
+		if err := service.SetPnpIDMacaddressMapping(); err != nil {
```

Comment:
What is the goal of refetching mappings here? and why error out if that fails?

Reply:
There can be case where ib's are not programmed during the start up. This is to address scenarios where it is not present at the boot up and available later.